### PR TITLE
Permission Grid: stick the headers to handle a lot of tags

### LINKED ISSUE #2810: Issue about bad visual if you have many tags permissions
## Bug Report

**Current Behavior**
A clear and concise description of the behavior.

**Steps to Reproduce**
1. Go to admin#/permissions
2. Here I have 8 long label tag that goes out of the screen
3. For be able to set permission and to see latest tab you need scroll down all the page than horizontal scrollbar is showed but is again hidden if you scroll app.

Is almost impossible set permission in this way because you cannot read the row: the left column has the text cut out impossible to read.

**Expected Behavior**

Read the left column and have a way to don't see the horizontal scrollbar and avoid seeing the horizontal bar disappear, if necessary. Maybe you just have to fix the fact description on the left are cut out of the screen for the right tag if you have many tags that need the horizontal scroll bar.

**Screenshots**

SharedScreenshot

Hard to understand permission in this way, maybe impossible!

**Environment**
- Flarum version: Beta 16
- Website URL: https://community.peopleinside.it/
- Webserver: [e.g. apache, nginx] -> Apache
- Hosting environment: [e.g. shared, vps] VPS
- PHP version:  7.4
- Browser: [e.g. chrome 67, safari 11] Firefox 89

```
Output of "php flarum info", run this in terminal in your Flarum directory.
```

Flarum core 0.1.0-beta.16
PHP version: 7.4.xx
Loaded extensions: Core, date, libxml, pcre, dom, fileinfo, filter, hash, json, SPL, bz2, posix, Reflection, session, SimpleXML, standard, xml, xmlreader, xmlwriter, bcmath, openssl, calendar, ctype, curl, dba, mbstring, ftp, gd, gettext, gmp, iconv, imap, intl, ldap, mailparse, exif, memcached, zlib, pcntl, PDO, pdo_sqlite, shmop, soap, sockets, sqlite3, sysvsem, sysvshm, tidy, tokenizer, xmlrpc, xsl, zip, Phar, mysqlnd, mysqli, pdo_mysql, ionCube Loader, Zend OPcache
+-------------------------------------+------------------+--------+
| Flarum Extensions                   |                  |        |
+-------------------------------------+------------------+--------+
| ID                                  | Version          | Commit |
+-------------------------------------+------------------+--------+
| flarum-flags                        | v0.1.0-beta.16   |        |
| flarum-markdown                     | v0.1.0-beta.16.1 |        |
| flarum-approval                     | v0.1.0-beta.16   |        |
| flarum-subscriptions                | v0.1.0-beta.16   |        |
| flarum-tags                         | v0.1.0-beta.16   |        |
| askvortsov-rich-text                | v1.1.1           |        |
| fof-linguist                        | 0.6.0            |        |
| fof-terms                           | 0.6.2            |        |
| fof-merge-discussions               | 0.5.1            |        |
| clarkwinkelmann-first-post-approval | 0.1.4            |        |
| fof-follow-tags                     | 0.7.2            |        |
| fof-username-request                | 0.5.0            |        |
| fof-split                           | 0.7.0            |        |
| fof-cookie-consent                  | 0.5.0            |        |
| fof-disposable-emails               | 0.3.0            |        |
| clarkwinkelmann-emojionearea        | 0.3.1            |        |
| fof-nightmode                       | 0.8.0            |        |
| fof-stopforumspam                   | 0.5.0            |        |
| fof-spamblock                       | 0.5.0            |        |
| fof-oauth                           | 0.3.0            |        |
| flarum-emoji                        | v0.1.0-beta.16   |        |
| fof-user-directory                  | 0.6.1            |        |
| flarum-pusher                       | v0.1.0-beta.16   |        |
| flarum-suspend                      | v0.1.0-beta.16   |        |
| fof-user-bio                        | 0.5.0            |        |
| fof-default-user-preferences        | 0.4.0            |        |
| nearata-lang-italian                | v0.1.0-beta.16   |        |
| flarum-akismet                      | v0.1.0-beta.16   |        |
| flarum-statistics                   | v0.1.0-beta.16   |        |
| flarum-lang-english                 | v0.1.0-beta.16   |        |
| flarum-sticky                       | v0.1.0-beta.16   |        |
| flarum-mentions                     | v0.1.0-beta.16   |        |
| flarum-lock                         | v0.1.0-beta.16   |        |
| flarum-likes                        | v0.1.0-beta.16   |        |
| flarum-bbcode                       | v0.1.0-beta.16   |        |
| noriods-auto-more                   | 0.3.1            |        |
| fof-links                           | 0.6.0            |        |
| fof-moderator-notes                 | 0.5.0            |        |
| fof-recaptcha                       | 0.3.2            |        |
| ianm-synopsis                       | 0.2.0            |        |
+-------------------------------------+------------------+--------+
Base URL: https://community.peopleinside.it

**Possible Solution**

Fix the left column text to be always visible or change the design

**Additional Context**

### PR DESCRIPTION
**Pull request by ornanovitch**
**Fixes flarum/framework#2810**
Temporary fix before flarum/issue-archive#253  

**Changes proposed in this pull request:**

  Stick `thead th` and `tbody th`: when you have a lot of tags permissions to handle, it's more consistent to be able to see the headers you're in. 

**Reviewers should focus on:**

To be tested with a lot of `Restrict by Tag` :label: 
Does it work on safari?

**Screenshot**

FA isn't loaded on my local dev environment, I don't know why but it has nothing to do with this PR :smile_cat:  
Screenshot_2021-05-25 forum

**Confirmed**

- [x] Frontend changes: tested on a local Flarum installation.

### CODE REVIEW

**Comment by SychO9** [comment]
Hello! 

Thank you for the PR, I have actually played with the idea a bit in the past, and really liked it.

We will review this soon when we pick up development again 🤗

**Comment by Ornanovitch** [comment]
Take your time, all of you deserve it! 

I'd be glad to start trying to improve a little Flarum with my limited skill but there is no rush at all, I'll wait for your review :)

**Comment by luceos** [comment]
@Ornanovitch thanks so much, this is awesome. I've assigned a couple of reviewers in the hope to push this into the next release 🤞 . The ux on the permissions page is seriously hampered with many tags or groups, so thank you for thinking along!

File: less/admin/PermissionsPage.less
Comment (SychO9) [inline less/admin/PermissionsPage.less:151]: It would cleaner to instead just add the normal bg color to the actual element `PermissionGrid-child` and not the th so that we don't have to use the `:not` here.

Additionally I'd also like to see that same bg color applied to `.PermissionGrid-section` so that the hidden tag names don't weirdly show up at times when scrolling down.

File: less/admin/PermissionsPage.less
Comment (SychO9) [inline less/admin/PermissionsPage.less:56]: I don't think this is necessary 🤔

File: less/admin/PermissionsPage.less
Comment (SychO9) [inline less/admin/PermissionsPage.less:69]: might be better without this as well 🤔

**Review by SychO9 — CHANGES_REQUESTED** [review]
Nice! a few comments

Can you add to

```less
.PermissionGrid-child {
  td, th {
    position: relative;
```

a `z-index: 0` as it can overlap the th when hovered.

File: less/admin/PermissionsPage.less
Comment (Ornanovitch) [reply less/admin/PermissionsPage.less:151]: > It would cleaner to instead just add the normal bg color to the actual element PermissionGrid-child and not the th so that we don't have to use the :not here

Oooops, obviously!

File: less/admin/PermissionsPage.less
Comment (Ornanovitch) [reply less/admin/PermissionsPage.less:56]: I didn't find another way to achieve the main point of this idea, i.e. stick the table heads. I don't know if `-webkit-sticky` is necessary as I don't really understand the safari world, but concerning the others, `position` and `top` are necessary to stick the columns headers, and `padding-bottom` is here to correct a side effect of them (the default `padding: 5px` seems very poor when scrolling down).

Maybe you're just suggesting to not stick the `thead th` and only stick the `tbody th`?

File: less/admin/PermissionsPage.less
Comment (Ornanovitch) [reply less/admin/PermissionsPage.less:69]: this is not strictly necessary but IMO there's a real aesthetic interest: without it, the `thead` row overlaps the permission labels column when scrolling horizontally while being on top

File: less/admin/PermissionsPage.less
Comment (SychO9) [reply less/admin/PermissionsPage.less:56]: my apologies, I was only referring to the bottom padding but for perhaps mistakenly selected all lines or smth. There is nothing wrong with the rest.

> and padding-bottom is here to correct a side effect of them (the default padding: 5px seems very poor when scrolling down)

makes sense 👍🏼

File: less/admin/PermissionsPage.less
Comment (SychO9) [reply less/admin/PermissionsPage.less:69]: There may have been some overlap before, but with the updated code, removing this works nicely, I think what I like about it is that it's no longer a dead area where neither the tags nor the permission labels take when scrolling.

See here what I mean:

https://user-images.githubusercontent.com/20267363/123166758-aafa8280-d46d-11eb-9042-932a8ccbd3e7.mp4

File: less/admin/PermissionsPage.less
Comment (Ornanovitch) [reply less/admin/PermissionsPage.less:56]: OK no problem!

File: less/admin/PermissionsPage.less
Comment (Ornanovitch) [reply less/admin/PermissionsPage.less:69]: OK I see the point! When scrolling down I totally agree, but when you're at the top and scrolling horizontally, aren't you annoyed by this?

| with `&:first-child`... | without `&:first-child`... |
| ------------------------- | --------------------------- |
| Firefox_Screenshot_2021-06-24T06-56-17 919Z | Firefox_Screenshot_2021-06-24T06-57-03 194Z |

It's a detail, maybe I'm too pernickety :shipit:

Comment (SychO9) [reply less/admin/PermissionsPage.less]: Yea, I think it's just a matter of preference really, I like being able to see the tags hidden when on top, but that might not be the case for others.

Let's just leave it open for a third opinion and we can make changes based on that!

Comment (Ornanovitch) [reply less/admin/PermissionsPage.less]: Roger that :+1:

Comment (davwheat) [reply less/admin/PermissionsPage.less]: Argh, it's a hard one! I like seeing the tag because it tells me there's more options to the left, but I dislike it because it doesn't look nice.

I'd err on the side of hiding it though, if I had to choose.

Comment (Ornanovitch) [reply less/admin/PermissionsPage.less]: > I like seeing the tag because it tells me there's more options to the left

Maybe that's the key for a temporary UX-based decision, even if I too dislike the design :see_no_evil: I'll try to play around the previous solution asap, and see if I can figure out a compromise

Comment (SychO9) [reply less/admin/PermissionsPage.less]: It's fine tbh, I'm not sure there can be a compromise, it's a small detail anyway, we can just bring it back 👍🏼

Comment (Ornanovitch) [reply less/admin/PermissionsPage.less]: Deal :slightly_smiling_face: 
I'll revert my last commit this evening (french time zone :clock10:) and I'll maybe play around a compromise for a future PR if I find a good idea.

Comment (Ornanovitch) [reply less/admin/PermissionsPage.less:69]: done :)

**Review by SychO9 — APPROVED** [review]
Looks good to me!

File: less/admin/PermissionsPage.less
Comment (askvortsov1) [inline less/admin/PermissionsPage.less:53]: I don't suppose we can avoid vendor specific stuff here?

**Review by askvortsov1 — APPROVED** [review]
Mostly LGTM, left a nit.

Is there a reason we haven't merged this yet, or have we just not gotten around to it yet?

File: less/admin/PermissionsPage.less
Comment (SychO9) [reply less/admin/PermissionsPage.less:53]: for now we need it for some old safari browsers

**Comment by SychO9** [comment]
> Is there a reason we haven't merged this yet, or have we just not gotten around to it yet?

Just needed a second approval!

**Pull request merged**
